### PR TITLE
feat: convert to 3-step stepper

### PR DESCRIPTION
**Pull request by raq929**
Converts to a 3-step stepper that is not in a fullscreen modal
Needs a Paragon upgrade to work correctly; that PR is in review: https://github.com/edx/paragon/pull/708
Marked as WIP because I'm working on tests.
Screen Shot 2021-05-07 at 9 32 43 AM

### CODE REVIEW

File: src/components/BulkEnrollmentPage/CourseSearchResults.jsx [hunk 1]
```diff
@@ -130,12 +137,6 @@ export const BaseCourseSearchResults = ({
   }
   return (
```
Comment (binodpant) [inline src/components/BulkEnrollmentPage/CourseSearchResults.jsx:138]: when I select all courses in the course selector stepper screen should I see an Enroll button also above the table to the right? I did not see it in the local version when I opened the first screen

File: src/components/BulkEnrollmentPage/stepper/ReviewStep.jsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { Row } from '@edx/paragon';
 import PropTypes from 'prop-types';
 
 import { BulkEnrollContext } from '../BulkEnrollmentContext';
```
[the remark this answers is not in the document]
Comment (raq929) [reply src/components/BulkEnrollmentPage/stepper/ReviewStep.jsx:5]: @alex-sheehan-edx is adding it as part of his ticket!

**Comment by raq929** [comment]
Oh, clear selection not clearing is definitely a bug; I'll look into it. Thanks!

File: __mocks__/react-instantsearch-dom.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
+/* eslint-disable object-curly-newline */
```
Comment (binodpant) [inline __mocks__/react-instantsearch-dom.jsx:1]: 🎉 this mock will be very handy I can see!

**Comment by binodpant** [comment]
I synced the latest on this branch and find that the 'Clear selection' does clear the checkbox state on the to pof the table, but not the checkbox in the row, here is what I see after clicking on 'Clear selection' in the learner screen

Screen Shot 2021-05-10 at 4 19 52 PM

Do you mind double checking on your end to see if I missed something?

I am looking through the rest of the PR but looking great

**Review by binodpant — COMMENTED** [review]
Once we verify clear selection issue, I think it's looking good to merge in to the feature branch

Tests are cool and appear to cover the functionality in detail

I presume the click enroll leaners functionality is part of Alex's PR so look forward to that. 

Also checked mobile screen view looking good.

**Comment by binodpant** [comment]
Also unclear to me if new in this branch but in stage I did not see this, is this styling something we could somehow have changed in this PR or obtained via a dependency update? It does not look like that, yet I see this

Screen Shot 2021-05-10 at 4 47 03 PM

we can probably check on this separately just adding here for completeness

**Comment by raq929** [comment]
@binodpant I suspect the styling error is introduced by paragon; it might be fixed by brand styles.

**Review by binodpant — COMMENTED** [review]
lgtm!

**Pull request merged**